### ISSUE TITLE
Warn on empty local assignment

### ISSUE THREAD

**Issue by rowlap** [koalaman/shellcheck#1297]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

set -u

myfunc() {
        local a=
        a=apple
        echo "a is $a"
}

myfunc
```

#### Here's what I wanted or expected to see:
shellcheck currently says nothing about this. The issue is that the empty assignment `local a=` defeats the purpose of `set -u`, which won't be triggered on empty string.

If the developer really means empty string this can be written as `local a=''`, otherwise `local a` is safer in that `set -u` is still effective against typos.

Other forms of assignment with modifiers are `alias a=b` / `declare a=b` / `typeset a=b` / `export a=b` / `readonly a=b` but I'm less sure that any of those are likely to appear with missing values.

**Comment by apenwarr** [comment]
Unfortunately a lot of shells, including dash, ksh, and busybox, do not seem to clear a automatically when you write "local a", only when you write "local a=".  So it would probably be dangerous to advise people to write it without the =.  In fact, maybe we should have the opposite warning.

**Comment by akinomyoga** [comment]
> In fact, maybe we should have the opposite warning.

Bash 5.0+ now has the shell options `shopt -s localvar_inherit` to make the behavior closer to some other shells, so the declarations without values generally cause undefined results. It would be useful to have the warning about the declarations in function scopes without values such as `local a`.
